x-pack/legacy/plugins/lens/public/indexpattern_plugin/operations/operations.ts

```diff
@@ -229,16 +231,26 @@ export function buildColumn({
     suggestedPriority,
     layerId,
     indexPattern,
+    previousColumn,
   };
 
   if (!field) {
     throw new Error(`Invariant error: ${operationDefinition.type} operation requires field`);
   }
 
-  return operationDefinition.buildColumn({
+  const newColumn = operationDefinition.buildColumn({
     ...baseOptions,
     field,
   });
+
+  if (
+    newColumn.dataType === 'number' &&
+    previousColumn?.params &&
+    'format' in previousColumn.params
+  ) {
+    newColumn.params = { format: previousColumn.params.format };
+  }
+  return newColumn;
```

Comment:
@timroes This is the number-specific code I was talking about

Reply:
What if we check that `newColumn.dataType === previousColumn.dataType`? We could say the formatters are specific to the data type